crates/autopilot/src/domain/competition/mod.rs

```diff
@@ -91,6 +91,35 @@ impl Score {
     }
 }
 
+#[derive(Clone)]
+pub struct RawParticipant {
+    pub driver: Arc<infra::Driver>,
+    pub solution: Solution,
+}
+
+#[derive(Clone)]
+pub struct Participant {
+    pub driver: Arc<infra::Driver>,
+    pub solution: Solution,
+    pub winner: bool,
+}
+
+/// Returns the order IDs of the winning solutions.
+pub fn winning_orders(participants: &[Participant]) -> impl Iterator<Item = &domain::OrderUid> {
```

Comment:
Unless this becomes relevant somewhere else I would only have these functions in the run loop where they are actually used.

Reply:
removed